src/RA_StringUtils.h

```diff
@@ -496,24 +461,31 @@ class StringBuilder
 
     bool m_bPrepareWide;
 
-    struct PendingString
+    class PendingString
     {
-        union {
-            const std::string* String;
-            const std::wstring* WString;
+    public:
+        // NB: Ref only does non corruption on the spans but corrupts pointers
+        // This is needed for a speed boost by constructing in place instead of copying or moving.
+        explicit PendingString(const std::string& arg) noexcept : Ref{arg}, DataType{Type::StringRef} {}
+        explicit PendingString(const std::wstring& arg) noexcept : Ref{arg}, DataType{Type::WStringRef} {}
+        explicit PendingString(std::string&& arg) noexcept : String{std::move(arg)}, DataType{Type::String} {}
+        explicit PendingString(std::wstring&& arg) noexcept : WString{std::move(arg)}, DataType{Type::WString} {}
+        explicit PendingString(std::string_view&& arg) noexcept : Ref{std::move(arg)}, DataType{Type::CharRef} {}
+        explicit PendingString(std::wstring_view&& arg) noexcept : Ref{std::move(arg)}, DataType{Type::WCharRef} {}
+        ~PendingString() noexcept = default;
 
-            struct
-            {
-                const char* Pointer;
-                size_t Length;
-            } Char;
+        // To prevent copying moving
+        PendingString& operator=(const PendingString&) = delete;
+        PendingString& operator=(PendingString&&) noexcept = delete;
 
-            struct
-            {
-                const wchar_t* Pointer;
-                size_t Length;
-            } WChar;
-        } Ref{};
+        // These are needed by m_vPending's allocator
+        PendingString(const PendingString&) = default;
+        PendingString(PendingString&&) noexcept = default;
+        PendingString() noexcept = delete;
+
+    private:
+        using RefType = std::variant<const std::string, const std::wstring, std::string_view, std::wstring_view>;
+        RefType Ref;
```

Comment:
I believe this creates copies of `std::string` and `std::wstring`. `RefType` should probably just be a variant of `string_view` and `wstring_view`, where they can be populated with the `c_str` and `length` of the source strings. Or the `std::string` and `std::wstring` variant parts should be references.

Reply:
I made it so string_view would do it for only pointers.
Originally I made the views for `StringRef `but it kept on being called incorrectly when they were supposed to be for `String` (`string_views `are non-owning references of strings so when called incorrectly it printed out a dangled string). After corrected, the `string_view` constructors weren't called at all, so I just removed them.